great_expectations/data_context/types/base.py

```diff
@@ -494,6 +560,8 @@ def __init__(
             self.spark_config = spark_config
         if boto3_options is not None:
             self.boto3_options = boto3_options
+        if azure_options is not None:
+            self.azure_options = azure_options
```

Comment:
`azure_options` is added wherever `boto3_options` are added just for parity. We can revisit whether or not objects like the `ExecutionEngine` are even utilizing these containers as a refactoring ticket.